src/main/java/com/kodilla/ecommercee/domain/Product.java

```diff
@@ -0,0 +1,53 @@
+package com.kodilla.ecommercee.domain;
+
+import com.kodilla.ecommercee.domain.dto.ProductsInCartDto;
+import lombok.*;
+import org.jetbrains.annotations.NotNull;
+
+import javax.persistence.*;
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.List;
+
+@Getter
+@Setter
+@NoArgsConstructor(force = true)
+@AllArgsConstructor
+@Entity
+@Table(name = "PRODUCTS")
+public class Product {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    @Column(name = "PRODUCT_ID", unique = true)
+    private Long productId;
+
+    @NotNull
+    @Column(name = "PRODUCT_NAME")
+    private String name;
+
+    @Column(name = "PRODUCT_DESCRIPTION", length = 1024)
+    private String description;
+
+    @NotNull
+    @Column(name = "PRODUCT_PRICE")
+    private BigDecimal price;
+
+    @ManyToOne(
+        fetch = FetchType.LAZY,
+        cascade = CascadeType.PERSIST)
+        @JoinColumn(name = "GROUP_ID")
+    private Group group;
+
+    @OneToMany(
+            mappedBy = "product",
+    fetch = FetchType.LAZY,
```

Comment:
I think that Martyna is correct, as relation is M:N and table `PRODUCTS_IN_CARTS` should be a table automaticly created by Hibernate no a sparate class
I belive that by using annotation `@ManyToMany` and `@JoinTable(
      name = "PRODUCTS_IN_CARTS",
      joinColumns = {@JoinColumn(name = "PRODUCT_ID", referencedColumnName = "PRODUCT_ID")},
      inverseJoinColumns = {@JoinColumn(name = "CART_ID", referencedColumnName = "CART_ID")}
)` 
Also i think we need check with _Andrzej_ how to implement productQuantity as it is neither in Product nor in Cart

Reply:
In the case of a @ManyToMany relationship, where many items from one table can refer to many items from another table, it can be difficult to manage, especially if we want to add additional information to that relationship, such as the quantity of items in the cart or the unit price.

In the case of an intermediate entity and a separate class for ProductsInCart, we can more easily add and manage additional information related to the items in the cart, as well as search and filter data more easily. For example, we can easily retrieve all products in the cart that cost more than a certain price or all carts that contain a specific product.